components/docker-auth/cmd/authn/authentication.go

```diff
@@ -105,15 +133,130 @@ func main() {
 	tokenValidity, err := validateToken(token, certificateInUse)
 	if err != nil {
 		glog.Errorf("Token is not valid - %s", err)
-		glogFlushAndExit(extension.ErrorExitCode)
+		extension.LogErrorAndExit(extension.ErrorExitCode)
 	}
 	glog.Info("signature verified")
 
 	if tokenValidity {
 		glog.Info("user successfully authenticated")
-		glogFlushAndExit(extension.SuccessExitCode)
+		extension.LogErrorAndExit(extension.SuccessExitCode)
 	} else {
 		glog.Error("authentication failed")
-		glogFlushAndExit(extension.ErrorExitCode)
+		extension.LogErrorAndExit(extension.ErrorExitCode)
+	}
+}
+
+// isJWT checks whether the token is jwt token or access token.
+func isJWT(conf extension.Conf) bool {
+	isJWTEnv := os.Getenv("IS_JWT")
+	var isJWT bool
+	if len(isJWTEnv) == 0 {
+		isJWT = conf.AuthConf.IsJWT
+	} else {
+		if isJWTEnv == "true" {
+			isJWT = true
+		} else if isJWTEnv == "false" {
+			isJWT = false
+		} else {
+			glog.Error("Wrong environment value given. The value should be either true or false")
+			extension.LogErrorAndExit(extension.ErrorExitCode)
+		}
+	}
+	return isJWT
+}
+
+// validateAccessToken is used to introspect the access token
+func validateAccessToken(token string, providedUsername string, conf extension.Conf) bool {
+	isHost, isPort := resolveISHostAndPort(conf)
```

Comment:
Better to rename these variables to `idpHost` and `idpPort` because the current names sounds like they are boolean values. Please search and fix in all places